subprojects/core-api/src/main/java/org/gradle/api/artifacts/Configuration.java

```diff
@@ -154,6 +155,16 @@ public String determineName(Configuration c) {
      */
     Configuration extendsFrom(Configuration... superConfigs);
 
+    /**
+     * Adds the given configuration to the set of configuration which this configuration extends from.
+     *
+     * @param superConfig The super configuration.
+     * @return this configuration
+     * @since 8.6
+     */
+    @Incubating
+    Configuration extendsFrom(Provider<? extends Configuration> superConfig);
```

Comment:
❓ What about `ProviderConvertible`? It may be better to introduce a`getParents(): SetProperty<Configuration>` to replace the `extendsFrom`. For now, the `extendsFrom` would simply contribute to the `SetProperty`. It avoids having to redefine all `add`/`addAll` API variants. Although `ProviderConvertible` is not yet accepted by `SetProperty`, I would assume at some point those will make their way to Gradle properties.